src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -15,13 +15,28 @@ namespace Microsoft.AspNetCore.Authentication
     public class AuthenticationSchemeProvider : IAuthenticationSchemeProvider
     {
         /// <summary>
-        /// Constructor.
+        /// Creates an instance of <see cref="AuthenticationSchemeProvider"/>
+        /// using the specified <paramref name="options"/>,
         /// </summary>
         /// <param name="options">The <see cref="AuthenticationOptions"/> options.</param>
         public AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options)
+            : this(options, new Dictionary<string, AuthenticationScheme>(StringComparer.Ordinal))
+        {
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="AuthenticationSchemeProvider"/>
+        /// using the specified <paramref name="options"/> and <paramref name="schemes"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="AuthenticationOptions"/> options.</param>
+        /// <param name="schemes">The dictionary used to store authentication schemes.</param>
+        protected AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options, IDictionary<string, AuthenticationScheme> schemes)
         {
             _options = options.Value;
 
+            _schemes = schemes ?? throw new ArgumentNullException(nameof(schemes));
```

Comment:
Personally I like this syntax, but some (including @Eilon I think) are zealously against using this syntax sugar... 

But maybe things have changed, I'd love to use this elsewhere :)

Reply:
@davidfowl has grudgingly accepted this.

Reply:
yay, means I can finally make the VS squiggles go away :)

Reply:
Naah I'm cool with it.